src/context/AnalyticsContext.tsx

```diff
@@ -0,0 +1,137 @@
+'use client' // Add this at the top of the file
```

Comment:
LGTM
Maybe just rm a few unused comments like this one. 
and line 3 in the same component.

Also, just to check - have we set a custom filter in the Google Analytics profile to limit the use of the unique Measurement ID to the production URL?